### PR TITLE
Reduce events on status changes #509

### PR DESCRIPTION
**Pull request by rigazilla**
This reduce to 2 the reconcile calls on cache creation

### CODE REVIEW

File: pkg/controller/cache/cache_controller.go [hunk 1]
```diff
@@ -206,11 +198,12 @@ func (r *ReconcileCache) Reconcile(request reconcile.Request) (reconcile.Result,
 		instance.Status.ServiceName = serviceList.Items[0].Name
 		statusUpdate = true
 	}
-	statusUpdate = statusUpdate || instance.SetCondition("Ready", metav1.ConditionTrue, "")
+	conditionUpdate := instance.SetCondition("Ready", metav1.ConditionTrue, "")
```
Comment (dmvolod) [inline pkg/controller/cache/cache_controller.go:201]: Oh, code optimizator prevents second expression to calculate it first time :)

**Comment by dmvolod** [comment]
But looks like we will add additional call with SetControllerReference update

File: pkg/controller/cache/cache_controller.go (hunk 1 above)
Comment (rigazilla) [reply pkg/controller/cache/cache_controller.go:201]: yep.
Well... I could probably just swap the `||` operands order

**Comment by rigazilla** [comment]
merged, thanks!

**Pull request merged**